daemon/export.go

```diff
@@ -61,8 +61,7 @@ func (daemon *Daemon) containerExport(container *container.Container) (arch io.R
 		}
 	}()
 
-	_, err = rwlayer.Mount(container.GetMountLabel())
-	if err != nil {
+	if err := daemon.Mount(container); err != nil {
```

Comment:
A couple of lines above we get the `rwlayer` (and below, we `rwlayer.Unmount()`. Should these become `daemon.Unmount()` instead?

Reply:
Everything seems to be correct as far as I can see (took me really long time to figure out), but the resulting code is confusing, as pointed out by @thaJeztah.  

Initially I wanted to switch to `daemon.Mount/Unmount()` but realized we need to use lower level logic to hold a reference to the layer in case container remove is attempted while export is in progress.

To make the code more readable, I suggest this patch instead:
```diff
--- a/daemon/export.go
+++ b/daemon/export.go
@@ -61,12 +61,12 @@ func (daemon *Daemon) containerExport(container *container.Container) (arch io.R
                }
        }()
 
-       _, err = rwlayer.Mount(container.GetMountLabel())
+       basefs, err := rwlayer.Mount(container.GetMountLabel())
        if err != nil {
                return nil, err
        }
 
-       archive, err := archivePath(container.BaseFS, container.BaseFS.Path(), &archive.TarOptions{
+       archive, err := archivePath(basefs, basefs.Path(), &archive.TarOptions{
                Compression: archive.Uncompressed,
                UIDMaps:     daemon.idMappings.UIDs(),
                GIDMaps:     daemon.idMappings.GIDs(),
```